src/libs/agent-runtime/anthropic/index.ts

```diff
@@ -0,0 +1,83 @@
+import Anthropic from '@anthropic-ai/sdk';
+import { AnthropicStream, StreamingTextResponse } from 'ai';
+import { ClientOptions } from 'openai';
+
+import { LobeRuntimeAI } from '../BaseAI';
+import { AgentRuntimeErrorType } from '../error';
+import { ChatCompetitionOptions, ChatStreamPayload, ModelProvider } from '../types';
+import { AgentRuntimeError } from '../utils/createError';
+import { debugStream } from '../utils/debugStream';
+import { handleOpenAIError } from '../utils/handleOpenAIError';
+
+export class LobeAnthropicAI implements LobeRuntimeAI {
+  private client: Anthropic;
+
+  constructor({ apiKey }: ClientOptions) {
+    if (!apiKey) throw AgentRuntimeError.createError(AgentRuntimeErrorType.InvalidAnthropicAPIKey);
+
+    this.client = new Anthropic({ apiKey });
+  }
+
+  async chat(payload: ChatStreamPayload, options?: ChatCompetitionOptions) {
+    const { messages, model, max_tokens, temperature, top_p } = payload;
+    const system_message = messages.find((m) => m.role === 'system');
+    const user_messages = messages.filter((m) => m.role !== 'system');
+
+    const requestPramas: Anthropic.MessageCreateParams = {
+      max_tokens: max_tokens || 1024,
+      messages: user_messages as Anthropic.Messages.MessageParam[],
+      model: model,
+      stream: true,
+      system: system_message?.content as string,
+      temperature: temperature,
+      top_p: top_p,
+    };
+
+    try {
+      const response = await this.client.messages.create(requestPramas);
+      const [prod, debug] = response.tee();
+
+      if (process.env.DEBUG_ANTHROPIC_CHAT_COMPLETION === '1') {
+        debugStream(debug.toReadableStream()).catch(console.error);
+      }
+
+      return new StreamingTextResponse(AnthropicStream(prod, options?.callback), {
+        headers: options?.headers,
+      });
+    } catch (error) {
+      if ('status' in (error as any)) {
+        switch ((error as Response).status) {
+          case 401: {
+            throw AgentRuntimeError.chat({
+              error: error as any,
+              errorType: AgentRuntimeErrorType.InvalidAnthropicAPIKey,
+              provider: ModelProvider.Anthropic,
+            });
+          }
+          case 403: {
```

Comment:
is all 403 error is a `LocationNotSupportError` error?

Reply:
@arvinxx 403 in anthropic API is a `permission_error`. Based on my own experience of using the API, most of the time (if not all) it just means LocationNotSupportError. Its API token doesn't have fine-grained role/permission control. Weighing between accuracy and user experience, I would suggest we treat it as location not supported right now (so users get clear instructions) until we have to separate it further. What do you say?

Reply:
I'm not sure whether an api key with not opened plan or insufficient balance will lead to a 403 error?

Reply:
@arvinxx hmm that's possible. I have an account that has free evaluation access. The API usage is free but throttled. I'm not sure whether "insufficient balance" will trigger this but it's possible. I will treat it as general biz error to start with then.

Reply:
@arvinxx I've removed the branch logic for 403. Will handle it as a general biz error for now.